### PR TITLE
test: improve branch coverage across modules

### PR DESCRIPTION
**Pull request by tkhquang**
## Summary

- Add 37 new tests (837 -> 874) targeting previously uncovered branches across 7 modules
- New `test_shutdown.cpp` for `DMK_Shutdown()` orchestration (was at 0% coverage)
- Append tests to existing files for HookManager (VMT edge cases, shutdown-then-operate, filtered hook IDs), Logger (async flush, invalid reconfigure, async shutdown), EventDispatcher (emit exception propagation), Profiler (accessor methods, file export verification), and input_codes (hash correctness, hex fallback, unknown enums)
- Update `docs/tests/README.md` project structure to list all 18 test files
- Update `AGENTS.md` testing section to document shutdown tests

## Test plan

- [x] All 874 tests pass on MinGW debug build
- [x] No flaky tests -- all new tests use deterministic assertions
- [x] No duplicate tests -- verified against existing test names
- [x] Existing tests unmodified -- new tests appended only

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage across event dispatching, hook management, input handling, logging, profiling, and shutdown semantics, including idempotence, hot-reload re-initialization, error-paths, and cleanup verification.
* **Documentation**
  * Updated test documentation and guidance to reflect the revised test lineup, reordered descriptions, and added shutdown orchestration guidance; removed the related-docs subsection.

### CODE REVIEW

File: tests/test_logger.cpp [hunk 1]
```diff
@@ -939,6 +939,138 @@ TEST_F(LoggerTest, StringToLogLevel_ConcurrentWithConfigure)
...
+    EXPECT_EQ(log_level_to_string(LogLevel::Debug), "DEBUG");
+    EXPECT_EQ(log_level_to_string(LogLevel::Info), "INFO");
+    EXPECT_EQ(log_level_to_string(LogLevel::Warning), "WARNING");
+    EXPECT_EQ(log_level_to_string(LogLevel::Error), "ERROR");
+}
+
+TEST_F(LoggerTest, Configure_AbsolutePath_Works)
+{
+    auto abs_log_file = std::filesystem::temp_directory_path() /
+                        ("test_logger_abspath_" + std::to_string(GetCurrentProcessId()) + ".log");
```
[the remark this answers is not in the document]
Comment (tkhquang) [reply tests/test_logger.cpp]: The two tests already run under the `LoggerTest` fixture (which owns the 
primary log file lifecycle). The secondary files created in these tests are 
test-specific artifacts that the fixture shouldn't own -- they verify 
reconfiguration to a *different* file. Added PID+counter for uniqueness. 
Moving them into a separate fixture would add complexity without benefit 
since cleanup is already handled with try/catch guards matching the 
pattern used elsewhere in this file (e.g., ConcurrentFileAccess tests).
